tests/python/relay/strategy/arm_cpu/test_generalized_conv2d.py

```diff
@@ -0,0 +1,122 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Helper class for testing variations of 2D convolution. Should be used by subclassing
+`GeneralizedConv2dTests`, and then setting the arguments using tvm.testing.parameter(s)."""
+
+from numpy.random import randint
+import tvm
+import tvm.testing
+from tvm import relay
+from tvm.testing.aot import AOTTestModel, compile_and_run, generate_ref_data
+from tvm.micro.testing.aot_test_utils import AOT_CORSTONE300_RUNNER
+from tvm.topi.utils import change_ndarray_layout
+
+
+class GeneralizedConv2dTests:
+    """Superclass which can be used to test regular, depthwise, or grouped conv2D. Cannot be used
+    for 5D data formats (NCHWc and such) as written, but could be extended. Might also be worth
+    abstracting some of this logic into an even more general class that could be used for other
+    operators.
+
+    Note that data_shape should always be a tuple of length four indicating the data shape in NHWC
+    format (it will later be reshaped according to the given data_layout), and kernel_size should be
+    a length two tuple giving the height and width of the kernel."""
+
+    @tvm.testing.requires_corstone300
+    def test_conv2d(
```

Comment:
No - the fixtures being undefined would raise an error. This is not run because of [Pytest's test discovery rules](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#test-discovery) - only classes starting with `Test` are considered to be tests.

This is why the abstract classes `GeneralizedConv2dTests`, `Conv2dTests`, `GroupConv2dTests`, etc. are _not_ run, but `TestGroupConv2d_NCHW_OIHW`, `TestDepthWiseConv2d_NCHWc` are. I've added a comment clarifying this.